plugin/evm/service.go

```diff
@@ -4,61 +4,82 @@
 package evm
 
 import (
+	"fmt"
 	"net/http"
-	"time"
 
 	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/set"
 )
 
 type ValidatorsAPI struct {
 	vm *VM
 }
 
+type GetCurrentValidatorsRequest struct {
+	NodeIDs []ids.NodeID `json:"nodeIDs"`
+}
+
 type GetCurrentValidatorsResponse struct {
 	Validators []CurrentValidator `json:"validators"`
 }
 
 type CurrentValidator struct {
-	ValidationID   ids.ID        `json:"validationID"`
-	NodeID         ids.NodeID    `json:"nodeID"`
-	Weight         uint64        `json:"weight"`
-	StartTimestamp uint64        `json:"startTimestamp"`
-	IsActive       bool          `json:"isActive"`
-	IsL1Validator  bool          `json:"isL1Validator"`
-	IsConnected    bool          `json:"isConnected"`
-	Uptime         time.Duration `json:"uptime"`
+	ValidationID     ids.ID     `json:"validationID"`
+	NodeID           ids.NodeID `json:"nodeID"`
+	Weight           uint64     `json:"weight"`
+	StartTimestamp   uint64     `json:"startTimestamp"`
+	IsActive         bool       `json:"isActive"`
+	IsL1Validator    bool       `json:"isL1Validator"`
+	IsConnected      bool       `json:"isConnected"`
+	UptimePercentage float32    `json:"uptimePercentage"`
```

Comment:
I think including both the raw uptime seconds and the percentage would be beneficial because validators will need to know the value in seconds for constructing an uptime message that peers will be willing to sign.

Reply:
It seemed  a bit redundant info as it can be calculable from the other. Only thing that might be changing is the lastUpdated/currentTime that's being used for percentage calculation. I added it nevertheless.